test/instruction_graph_memory_tests.cc

```diff
@@ -1059,3 +1090,45 @@ TEST_CASE("device kernels report global memory traffic estimate based on range m
 		CHECK(reduce->estimated_global_memory_traffic_bytes == buffer_size / num_devices * sizeof(float));
 	}
 }
+
+TEST_CASE("5-point stencil program with 2D split does not exchange boundaries with diagonal neighbor devices",
```

Comment:
I guess we should also have a CDAG test:

```c++
TEST_CASE("5-point stencil program with 2D split does not exchange boundaries with diagonal neighbor nodes", "[command_graph_generator][command-graph]") {
	cdag_test_context cctx(4 /* num nodes */);

	constexpr range<2> buffer_size(256, 256);
	auto buf_a = cctx.create_buffer<2>(buffer_size);

	cctx.device_compute(range(buffer_size)).name("init").hint(experimental::hints::split_2d()).discard_write(buf_a, acc::one_to_one()).submit();
	cctx.device_compute(range(buffer_size))
	    .name("read")
	    .hint(experimental::hints::split_2d())
	    .read(buf_a, acc::neighborhood(range(1, 1), acc::along_axes))
	    .submit();

	const std::vector<std::pair<node_id, node_id>> diagonals{{0, 3}, {1, 2}, {2, 1}, {3, 0}};
	for(auto [nid, diag] : diagonals) {
		CHECK(std::ranges::none_of(cctx.query<push_command_record>().on(nid).assert_count(1)->target_regions, [&](auto& tr) { return tr.first == diag; }));
	}
}
```

Reply:
Added this.